### PR TITLE
parser: support to character option to load data statement

### LINKED ISSUE #7348: LOAD DATA does not support to set character set
Part of MySQL syntax looks like below:
```
LOAD DATA [LOW_PRIORITY | CONCURRENT] [LOCAL] INFILE 'file_name'
    [REPLACE | IGNORE]
    INTO TABLE tbl_name
    [PARTITION (partition_name [, partition_name] ...)]
    [CHARACTER SET charset_name]
```
It has a character set option in it, but TiDB not.
TiDB version: 0d18ac2c18c31d7d2fcf85ab7828359df0408c7f

### PR DESCRIPTION
**Pull request by imtbkcat**
### What problem does this PR solve? 
Fix #7348  to make some JDBC test cases happy.

### What is changed and how it works?
Add a CNF grammar to support `CHARACTER SET <charset_name>` at the end of `LOAD DATA`.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported variable/fields change

Side effects

no

Related changes

### CODE REVIEW

File: parser/parser.y [hunk 1]
```diff
@@ -6821,7 +6821,7 @@ RevokeStmt:
  * See https://dev.mysql.com/doc/refman/5.7/en/load-data.html
  *******************************************************************************************/
 LoadDataStmt:
-	"LOAD" "DATA" LocalOpt "INFILE" stringLit "INTO" "TABLE" TableName Fields Lines ColumnNameListOptWithBrackets
+	"LOAD" "DATA" LocalOpt "INFILE" stringLit "INTO" "TABLE" TableName Fields Lines ColumnNameListOptWithBrackets CharsetOpt
```
Comment (jackysp) [inline parser/parser.y]: CharsetOpt should behind TableName.

**Comment by imtbkcat** [comment]
PTAL @jackysp

File: parser/parser.y [hunk 2]
```diff
@@ -6821,25 +6821,29 @@ RevokeStmt:
  * See https://dev.mysql.com/doc/refman/5.7/en/load-data.html
  *******************************************************************************************/
 LoadDataStmt:
-	"LOAD" "DATA" LocalOpt "INFILE" stringLit "INTO" "TABLE" TableName Fields Lines ColumnNameListOptWithBrackets
+	"LOAD" "DATA" LocalOpt "INFILE" stringLit "INTO" "TABLE" TableName CharsetOpt Fields Lines ColumnNameListOptWithBrackets
```
Comment (jackysp) [inline parser/parser.y]: Please remove the last space.

**Pull request merged**